packages/core/strapi/lib/types/core-api/controller.d.ts

```diff
@@ -0,0 +1,52 @@
+import type { Common, CoreApi, Utils } from '@strapi/strapi';
+import type { ExtendableContext } from 'koa';
+
+/**
+ * Base Core-API controller type
+ *
+ * TODO: Make use of the T generic to type the other methods based on the given content type
+ */
+export interface Base {
+  // TODO: Use actual entities instead of regular object
```

Comment:
Are these TODOs planned needed for this PR or will they come in a later iteration?

Reply:
In a later iteraction, once we have the controller registries working